Added camel case aliases for all public methods and exported functions

sampeka opened the pull request:
As per #3. 

Also included some test coverage for the added helpers.

sampeka commented:
It's failing in node 0.12 because `string.startsWith` isn't there. Is there a way we can just configure babel to handle that? Or would you rather I just rewrite it slightly? Not a big deal either way.

halt-hammerzeit commented:
Hello.
I looked through your Pull Request and I understand only 
```
aliasCamelCaseAttrs(webpack_isomorphic_tools.prototype)
```
and
```
function aliasCamelCaseAttrs() { ... }
function camelCase() { ... }
```
I don't see the purpose of all the other code (except tests)

sampeka commented:
The rest just makes camel case aliases for all exported functions and adds them to `module.exports`. It's not totally necessary to have `helpers` and the other modules also export camel case, but it makes it more consistent if somebody does for whatever reason decide to use any of them in their own code. If you disagree though I'll strip the PR back to just `index.js`

halt-hammerzeit commented on source/common.js:
```diff
@@ -278,3 +278,5 @@ export function uniform_path(asset_path)
 	return asset_path
 }
+
+exportAllCamelCase(module);
```
is it required for any use case?

halt-hammerzeit commented on source/plugin/notify stats.js:
```diff
@@ -51,4 +52,6 @@ export default function notify_stats(stats, json, verbose)
\ No newline at end of file
+}
+
+exportAllCamelCase(module)
```
is it required for any use case?

sampeka replied:
I guess not ;) I'll remove the module-level stuff

sampeka commented:
Done, removed the module-level stuff. Also added an extra test to make sure the camel case attributes have been properly added to the isomorpher prototype. If you're happy, let me know and I'll squash the commits.

halt-hammerzeit commented:
@sampeka ok, that looks good to go.
I'll accept the PR after you squash it

sampeka commented:
Squashed, also just realised that people use `plugin.js` directly as well. Not all of the methods had camel case variants, so did the same there + added some light coverage.

halt-hammerzeit commented:
Ok, your contribution accepted.
i'll go release a new version.

halt-hammerzeit commented:
done
`webpack-isomorphic-tools@2.2.44`

halt-hammerzeit commented:
FYI
I'm posting this in every issue and PR to notify whoever may be interested:
today I've released an alternative helper library called [`universal-webpack`](https://github.com/halt-hammerzeit/universal-webpack).
It takes a different approach than `webpack-ismorphic-tools` and instead of hacking Node.js `require()` calls it just compiles all code with `target: 'node'` webpack configuration option.
As a result, all Webpack plugins and features are supported.
If you think you might need that here's an example project:
https://github.com/halt-hammerzeit/webpack-react-redux-isomorphic-render-example
